src/optimizer.cc

```diff
@@ -523,12 +523,25 @@ Grammar Optimizer::optimize() {
             log(0, "Running debug script %s...", debug_script.c_str());
             std::string filename = TempDir::get("pass_" + std::to_string(pass) + ".peg");
             write_file(filename, g.to_string());
-            int exit_code;
-            exit_code = WEXITSTATUS(system((debug_script + " " + filename).c_str()));
-            if (exit_code != 0) {
-                error("Debug script failed in pass %d! (exit_code=%d)", pass, exit_code);
+
+            int ret = system((debug_script + " " + filename).c_str());
+
+            if (ret == -1) {
+                perror(NULL);
+                error("Debug script failed in pass %d", pass);
```

Comment:
I think we could log `errno` here, to indicate why did the script failed. Something like 
```suggestion
                error("Debug script failed in pass %d: %s", pass, strerror(errno));
```

Reply:
I have invoked `perror(NULL);` before this line. Do you mean remove `perror(NULL);` and use `strerror(errno)`?

Reply:
Sorry, I've overlooked that. But yes, I think that printing both in one line is slightly more readable:

```
No such file or directory
ERROR: Debug script failed in pass 1
```
vs.
```
ERROR: Debug script failed in pass 1: No such file or directory
```